### ISSUE TITLE
no instance of function template "EntitasPP::ComponentTypeId::Get" matches the argument list

### ISSUE THREAD

**Issue by RvRooijen** [JuDelCo/Entitas-Cpp#3]
```
shared_ptr<Group> _group;

public:
    void SetPool(Pool* pool) override
    {
        _group = pool->GetGroup(Matcher_AllOf(ObjectmatrixComponent, PositionComponent));
```

When i want to get a group with a matcher that has multiple types i get this error.
no instance of function template "EntitasPP::ComponentTypeId::Get" matches the argument list
The types are both IComponent. 

I also tried it like this.

```
ComponentIdList l;

public:
    void SetPool(Pool* pool) override
    {
        l.push_back(ComponentTypeId::Get<ObjectmatrixComponent>());
        l.push_back(ComponentTypeId::Get<PositionComponent>());

        Matcher m = Matcher::AllOf(l);
        pool->GetGroup(m)->OnEntityAdded += [](std::shared_ptr<Group> group, EntityPtr entity, ComponentId index, IComponent* component)
        {
            cout << "test" << endl;
        };
```

Then i get a crash.

void Pool::UpdateGroupsComponentAddedOrRemoved
events has a nullptr.

**Comment by JuDelCo** [comment]
Hi,

As I said in #2 , at this moment I don't have time to maintain the lib and i'm well aware of those problems, the library has some memory leaks which needs to be fixed asap, but again, i'm very sorry i can't fix them atm.

I don't know when i will have again free time to spend in this library, but my current Job is taking up all my time u__u

I hope will be soon (in a couple of months), but I can't make any promises.

Meanwhile, you are free to fix yourselves the lib making a fork ;)

PS: As i said in #2, i will notify you when i'm back and fixed this library

**Comment by JuDelCo** [comment]
Fixed in bf5653e8bf41b06f23056876756dd7334d6d211b.

**NOTE**: You need now to use a `weak_ptr` for Groups in your systems.
